test/e2e/space_autocompletion_test.go

```diff
@@ -0,0 +1,157 @@
+		ManuallyApprove().
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		EnsureMUR().
+		Execute()
+	NewSignupRequest(t, awaitilities).
+		Username("for-member2").
+		Email("for-member2@redhat.com").
+		TargetCluster(memberAwait2).
+		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
+		ManuallyApprove().
+		EnsureMUR().
+		Execute()
+	hostAwait.UpdateToolchainConfig(testconfig.Tiers().DefaultSpaceTier("appstudio"))
+
+	t.Run("set low capacity threshold and expect that space will have default tier, but won't have target cluster so it won't be provisioned", func(t *testing.T) {
+		// given
+		hostAwait.UpdateToolchainConfig(
+			testconfig.AutomaticApproval().ResourceCapacityThreshold(1))
+
+		// when
+		space := CreateSpace(t, awaitilities, WithTierName(""))
+
+		// then
+		space = spaceIsPendingCluster(t, hostAwait, space.Name)
+		assert.Equal(t, "appstudio", space.Spec.TierName)
+
+		t.Run("reset the threshold and expect the space will be have the targetCluster set and will be also provisioned", func(t *testing.T) {
+			// when
+			hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().ResourceCapacityThreshold(80))
+
+			// then
+			VerifyResourcesProvisionedForSpace(t, awaitilities, space)
+		})
+	})
+
+	t.Run("set low max number of users and expect that space won't be provisioned but added on waiting list", func(t *testing.T) {
+		// given
+		toolchainStatus, err := hostAwait.WaitForToolchainStatus(wait.UntilToolchainStatusHasConditions(ToolchainStatusReadyAndUnreadyNotificationNotCreated()...))
+		require.NoError(t, err)
+		originalMursPerDomainCount := toolchainStatus.Status.Metrics[toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey]
+		hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().MaxNumberOfUsers(originalMursPerDomainCount["internal"] + originalMursPerDomainCount["external"]))
+
+		// when
+		space1 := CreateSpace(t, awaitilities, WithName("space-waitinglist1"))
+
+		// we need to sleep one second to create UserSignup with different creation time
+		time.Sleep(time.Second)
+		space2 := CreateSpace(t, awaitilities, WithName("space-waitinglist2"))
+
+		// then
+		spaceIsPendingCluster(t, hostAwait, space1.Name)
+		spaceIsPendingCluster(t, hostAwait, space2.Name)
+
+		t.Run("increment the max number of users and expect the both of the spaces will be provisioned. When we count the spaces, then this test will change", func(t *testing.T) {
+			// when
+			hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().MaxNumberOfUsers(originalMursPerDomainCount["internal"] + originalMursPerDomainCount["external"] + 1))
+
+			// then
+			VerifyResourcesProvisionedForSpace(t, awaitilities, space1)
+			VerifyResourcesProvisionedForSpace(t, awaitilities, space2)
+			// TODO: when we count the number of provisioned spaces, then the second space won't be provisioned immediately https://issues.redhat.com/browse/CRT-1427
+			// spaceIsPendingCluster(t, hostAwait, space2.Name)
```

Comment:
do you want to keep the commented out code here?

Reply:
that was my intention - as soon as we migrate metrics/capacity management then we can uncomment this part.
if it's noisy, then I can remove it

Reply:
It's fine, we can keep it: my comment was more a reminder than a request ;)